src/Sylius/Bundle/CoreBundle/Fixture/Factory/OrderExampleFactory.php

```diff
@@ -183,18 +183,20 @@ protected function createOrder(ChannelInterface $channel, CustomerInterface $cus
     protected function generateItems(OrderInterface $order): void
     {
         $numberOfItems = random_int(1, 5);
-        $products = $this->productRepository->findAll();
+        $channel = $order->getChannel();
+        $products = $this->productRepository->findByChannel($channel, 100);
```

Comment:
Instead of adding `findByChannel` method, we can use `findBy(['channel' => $channel], [], 100)` here.

Reply:
Therefore we won't need that method and we can push it as a bugfix :)

Reply:
There is no `channel` prop though, it's `channels`. We need `MEMBER OF`

Reply:
You're right, we either need to find some other way around it using the existing methods or this PR needs to go to the master branch.

Reply:
How about:

```
        $channel = $order->getChannel();
        $locale = $channel->getDefaultLocale();
        $products = $this->productRepository->findLatestByChannel($channel, $locale->getCode(), 100);
        if (0 === count($products)) {
            throw new \InvalidArgumentException(sprintf(
                'You have no enabled products at the channel "%s", but they are required to create an orders for that channel',
                $channel->getCode()
            ));
        }
```
(here we stick on enabled products only, but that's ok I guess)

Reply:
Force-pushed this change.